src/core/Target.mli

```diff
@@ -0,0 +1,140 @@
+  products : Semgrep_output_v1_t.product list;
+      (** The products which should scan this target. This is used for
+          selecting the relevant set of rules. *)
+  lockfile : lockfile option;
+      (** Optional lockfile associated with this target.
+
+          The association is namely that this target has its dependencies
+          specified by the given lockfile. Core doesn't need to worry about
+          determining these associations; rather, the target selection and
+          generation process must resolve these connections as part of
+          generating regular targets. *)
+}
+[@@deriving show]
+(** A regular semgrep target, comprising source code (or, for
+   regex/generic, arbitrary text data) to be executed. See also {!Xtarget.t},
+   an augmented version which also has the contents. *)
+
+(** A Semgrep target. This contains all of the details needed to be able to
+    determine how to scan a target, e.g.,
+
+    {ul
+      {- What products should we select rules from?}
+      {- Where can we get the contents of the target?}
+      {- What language should we analyze the target as?}
+    }
+
+    However, it does not contain the actual contents (parsed or otherwise) of
+    the target itself. For that, see {!Xtarget.t} or {!Lockfile_xtarget}.
+ *)
+type t = Regular of regular | Lockfile of lockfile [@@deriving show]
+
+val mk_regular :
+  ?lockfile:lockfile ->
+  Xlang.t ->
+  Semgrep_output_v1_t.product list ->
+  Origin.t ->
+  regular
+(** [mk_regular analyzer products origin] is a {!regular} target
+      originating from [origin] to be analyzed with [analyzer] for [products].
+      If [lockfile] is specified then it shall be used as the associated
+      lockfile if dependency patterns are to be ran.
+
+      This function should be generally preferred over creating a record
+      directly, since it can peform actions which may be required when creating
+      a target from certain types of origins, such as generating a tempfile.
+ *)
+
+val mk_lockfile : ?manifest:manifest -> Lockfile_kind.t -> Origin.t -> lockfile
+(** [mk_lockfile k origin] is the a {!lockfile} target
+      originating from [origin] of kind [k]. If [manifest] is specified, it
+      shall be used as the associated manifest.
+
+      This function should be generally preferred over creating a record
+      directly, since it can peform actions which may be required when creating
+      a target from certain types of origins, such as generating a tempfile.
+ *)
+
+val mk_manifest : Manifest_kind.t -> Origin.t -> manifest
+(** [mk_manifest k origin] is a {!manifest} target
+    originating from [origin] of kind [k].
+
+    This function should be generally preferred over creating a record
+    directly, since it can peform actions which may be required when creating a
+    target from certain types of origins, such as generating a tempfile.
+ *)
+
+val internal_path_to_content : t -> Fpath.t
```

Comment:
rename just path (or fpath), otherwise call sites are harder to read.
(you can keep the internal_path_to_content field name though)

Reply:
How's `internal_path`? I would prefer something which at least indicates this may not be meaningful to the user, since I could see people potentially using `path` without thinking about it.

Reply:
yes, I like internal_path

Reply:
Done in d57830ad536d2800a7e4c53159e6c156191eae83.